coderdojochi/views.py

```diff
@@ -400,8 +400,9 @@ def session_sign_up(request, year, month, day, slug, session_id, student_id=Fals
                 session_obj.mentors.add(mentor)
         else:
             if user_signed_up:
-                order = get_object_or_404(Order, student=student, session=session_obj)
-                order.delete()
+                order = get_object_or_404(Order, student=student, session=session_obj, active=True)
```

Comment:
I think for this query you can leave out active=True because whether or not it was set properly to active=True, you want to change it to False.  Just in case for some reason active is unset, they will still be marked inactive rather than a 404. So it would be this...

` order = get_object_or_404(Order, student=student, session=session_obj)`